### PR TITLE
Do not set UPE billing details when changing a subscription payment method

### LINKED ISSUE #2917: Unable to change a subscription payment method via My Account
### Describe the bug
The following error occurs when attempting to change a subscription's payment method via My Account when UPE is enabled.

> You specified "never" for fields.billing_details.name when creating the payment Element, but did not pass confirmParams.payment_method_data.billing_details.name when calling stripe.confirmPayment or stripe.confirmSetup. If you opt out of collecting data via the payment Element using the fields option, the data must be passed at confirm-time.

Screen Shot 2021-09-15 at 3 43 32 pm

### To Reproduce

1. As a Merchant, ensure UPE checkout is enabled.
1. As a Shopper, purchase any subscription product.
1. Navigate to My account -> Subscriptions and open the newly created subscription entry.
1. Observe 3 buttons for “Actions”. Click “Change payment” button.
1. In the second block, select “Use a new payment method” option.
1. Provide 4242424242424242 as the test card number, future expiration date, and a CVC code.
1. Click “Change payment method” button.
1. Observe the above error.

### Actual behavior
An error is presented when attempting to change a subscription's payment method via My Account when UPE is enabled.

### Expected behavior
I am able to successfully change a subscription's payment method via My Account when UPE is enabled.

### Desktop (please complete the following information):

* OS: MacOS Big Sur
* Browser: Chrome
* Version: Version 92.0.4515.107 (Official Build) (x86_64)

### PR DESCRIPTION
**Pull request by aprea**
Fixes #2917

#### Changes proposed in this Pull Request

Tweaks a condition in `upe.js` to ensure the `billingDetails` UPE config is not set when changing payment methods for an existing subscription.

Prior to this change, attempting to change the payment methods for an existing subscription would result in the following error when entering new card details.

Screen Shot 2021-09-15 at 3 43 32 pm

#### Testing instructions

* As a Merchant, ensure UPE checkout is enabled and the WooCommerce Subscriptions plugin is installed. 
* As a Shopper, purchase any subscription product.
* Navigate to **My account → Subscriptions** and open the newly created subscription entry.
* Observe 3 buttons for “Actions”. Click “Change payment” button.
* In the second block, select “Use a new payment method” option.
* Provide `4242424242424242` as the test card number, future expiration date, and a CVC code.
* Click the “Change payment method” button.
* Ensure the payment method is updated successfully.

-------------------

- [x] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: client/checkout/classic/upe.js [hunk 1]
```diff
@@ -273,6 +273,7 @@ jQuery( function ( $ ) {
 		// If paying from order, we need to create Payment Intent from order not cart.
 		const isOrderPay = getConfig( 'isOrderPay' );
 		const isCheckout = getConfig( 'isCheckout' );
+		const isChangingPayment = getConfig( 'isChangingPayment' );
```
Comment (chickenn00dle) [inline client/checkout/classic/upe.js:276]: This is a nitpick and not a blocker. 

I noticed this is defined a few times in this file. Is there a reason we can't define this once in the outermost scope of the main function?

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Nice work @aprea!

I tested and this resolves the error that appears on the change payment method 🙌 

I just had one nitpick that can be ignored. Other than that, it looks like there are still failing tests which need to be fixed.

**Comment by aprea** [comment]
Looks like the "Compatibility - WC, WP and PHP / WC compatibility (4.0.0, 5.4, 7.0) (pull_request)" check will continue failing until https://github.com/Automattic/woocommerce-payments/pull/2945 is merged

**Comment by chickenn00dle** [comment]
Ah. Good catch. In that case, this LGTM 👍

**Review by chickenn00dle — APPROVED** [review]

**Pull request merged**